src/basic/architecture.h

```diff
@@ -237,6 +238,9 @@ int uname_architecture(void);
 #  else
 #    error "Unrecognized riscv architecture variant"
 #  endif
+#elif defined(__arc__)
+#  define native_architecture() ARCHITECTURE_ARC
+#  define LIB_ARCH_TUPLE "arc-linux"
```

Comment:
Well the thing is that's the most flexible option for ARC.
ARC toolchain usually comes from either release archives we prepare (see https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/) or from buildsystems such as Buildroot.

So what we have in our prebuilt tools are:
 * arc-linux
 * arc-linux-uclibc
 * arc-snps-linux-uclibc

That's in Buildroot:
 * arc-linux
 * arc-buildroot-linux-gnu

That might not indeed conform requirements of big distros but given we don't have those distros for ARC yet I think we may live like that. Makes any sense?